crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -10,23 +14,61 @@ use super::{
     tx_digest::TransactionDigest,
 };
 use async_graphql::*;
+use sui_json_rpc_types::{
+    SuiExecutionStatus, SuiTransactionBlockDataAPI, SuiTransactionBlockEffects,
+    SuiTransactionBlockEffectsAPI,
+};
 
-#[derive(SimpleObject, Clone, Eq, PartialEq)]
+#[derive(SimpleObject, Clone)]
+#[graphql(complex)]
 pub(crate) struct TransactionBlock {
     pub digest: TransactionDigest,
     pub effects: Option<TransactionBlockEffects>,
     pub sender: Option<Address>,
     pub bcs: Option<Base64>,
     pub gas_input: Option<GasInput>,
-    pub expiration: Option<Epoch>,
 }
 
-#[derive(SimpleObject, Clone, Eq, PartialEq)]
+impl From<sui_json_rpc_types::SuiTransactionBlockResponse> for TransactionBlock {
```

Comment:
Consider adding an import of `SuiTransactionBlockResponse` instead of the qualified version here and on the next line.